Beginner Guide Submission (#653)
OWASP/wrongsecrets: CONTRIBUTING.md

-   Unnecessary self-promotion of tools or blog posts is frowned upon. If you have a relation with on of the URLs or tools you are referencing, please state so in the PR so that we can verify that the reference is in line with the rest of the guide.

Please be sure to take a careful look at our [Code of Conduct](https://github.com/OWASP/wrongsecrets/blob/master/CODE_OF_CONDUCT.md) for all the details.

---
# Beginner guide
## OWASP WrongSecrets

> Basically, [*WrongSecrets*](https://owasp.org/www-project-wrongsecrets/) is an application which teaches how to not store secrets by offering challenges to the user, which helps the user to Self-reflect and correct those mistakes.

Reviewer: ```suggestion
[*WrongSecrets*](https://owasp.org/www-project-wrongsecrets/) is an application teaching how to *not* store secrets by offering challenges to the user, helping the user to Self-reflect and correct those mistakes.
```

      ![](images/lombok-setup-4.2.png)

    - Make sure that the ***Lombok processing*** is enabled.
      ![](images/lombok-processing-4.3.png)

- ### Step 5: Reload the project
    - Open the ***Maven*** Tab
    
      ![](images/open-maven-5.1.png)

    - Press the ***Reload*** button as shown below and allow the project to Reload.
    
      ![](images/reload-maven-5.2.png)

- ### Step 6: Running the Project.
    - Open the ***WrongSecretsApplication*** by following the path ***main>java>org.owasp.wrongsecrets>WrongSecretApplication***.
      ![](images/open-application-6.1.png)
    - Press ***Shift+F10*** to run the application, this will open up the ***Run/Debug Configurations Menu.***
      ![](images/run-application-6.2.png)

- ### Step 7: Setting up Configurations.
    -  Select ***Edit configuration templates*** then select ***Application*** section.
    
       ![](images/edit-config-7.1.png)
    - There under the ***Application*** section click on the button shown below.
    
      ![](images/modify-options-7.2.png)
    - ***Select*** all the fields that are Selected in the below picture.
    
      ![](images/select-options-7.3.png)

>     - ***Fill*** all the fields as shown below.

Reviewer: ```suggestion
    - ***Fill out*** all the fields as shown below.
```

    
      ![](images/fill-fields-7.4.png)
    - Again press ***Shift+F10*** which runs the Application.
    
      ![](images/run-application-6.2.png)

- ### There you have it, ***WrongSecrets*** running successfully.
    - Here is a *preview* on how does it look after successfully running the Application.
      **Note:** Running the Application doesn't open any kind of ***GUI***, it only initializes the ***local webserver*** that you can open via a ***browser.***
      ![](images/final-output-8.png)

    - Here is the preview of the **web server**,  you can try to find the secrets by means of solving the challenge offered at:
      [**Challenges**](https://github.com/OWASP/wrongsecrets#basic-docker-exercises)
        ![](images/screenshot.png)

---
## How to add challenge
- ### Step 1: Creating a new issue.

>   First make sure that you have an [Issue](https://github.com/OWASP/wrongsecrets/issues/new) reported for which a challenge is really wanted, And make sure the challenge is assigned to you, as others might be working at the challenge.

Reviewer: ```suggestion
  First make sure that you have an [Issue](https://github.com/OWASP/wrongsecrets/issues/new) reported for which a challenge is really wanted, And make sure the challenge is assigned to you, as others might be working on the challenge.
```

  Add the **new TestFile** in this folder `wrongsecrets/src/test/java/org/owasp/wrongsecrets/challenges/`. TestFile is required to do **unit testing.**
  These are the things that you have to keep in mind.
    - Make sure that this file is also of **Java** type.
    - Here is a unit test for reference:
        ```java
        package org.owasp.wrongsecrets.challenges.docker;
        import org.assertj.core.api.Assertions;
        import org.junit.jupiter.api.Test;
        import org.junit.jupiter.api.extension.ExtendWith;
        import org.mockito.Mock;
        import org.mockito.Mockito;
        import org.mockito.junit.jupiter.MockitoExtension;
        import org.owasp.wrongsecrets.ScoreCard;
        @ExtendWith(MockitoExtension.class)
        class Challenge28Test {
            @Mock
            private ScoreCard scoreCard;
            @Test
            void rightAnswerShouldSolveChallenge() {
                var challenge = new Challenge28(scoreCard);
                Assertions.assertThat(challenge.solved("wrong answer")).isFalse();
                Assertions.assertThat(challenge.solved(challenge.spoiler().solution())).isTrue();
            }
        }
        ``` 

>   Please note that PRs for new challenges are only excepted when unit tests are added to prove that the challenge works. Normally tests should not immediately leak the actual secret, so leverage the `.spoil()` functionality of your test implementation for this.

Reviewer: ```suggestion
  Please note that PRs for new challenges are only accepted when unit tests are added to prove that the challenge works. Normally tests should not immediately leak the actual secret, so leverage the `.spoil()` functionality of your test implementation for this.
```

- ### Step 4: Adding explanations, reasons and hints.
  Add the explanation for your challenge along with the hints that will help in finding the secret in this folder `wrongsecrets/src/main/resources/explanations/`.
  Things to be noted.
    - All the possible explanations for your challenge, included with all the hints and reasons should be provided.

>     - Every thing must be in a separate **AsciiDoc files**.

Reviewer: ```suggestion
    - Everything must be in separate **AsciiDoc files**.
```
